### PR TITLE
⚠ Add testenv support for CRDv1/CRDv1beta1 + update to k8s 1.16.4

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

This PR adds support for testenv to load CRDv1 and CRDv1beta1 by using `runtime.Object` and controller-runtime unstructured client.

### CODE REVIEW

**Comment by vincepri** [comment]
The failure seems like a race condition, it doesn't happen locally. I can take a look after getting some feedback on the current approach

**Comment by gerred** [comment]
@vincepri looks reasonable, I will test locally as well, but kicking the flake for now.

**Comment by vincepri** [comment]
@gerred @DirectXMan12 Thoughts on how to proceed here?

**Comment by DirectXMan12** [comment]
looks like it's passing.  Will defer to @gerred for the LGTM.  Let me know when that's in place

**Comment by gerred** [comment]
In transit at the moment but I will review again and lgtm here (or have any last feedback) in an hour or two!

**Comment by gerred** [comment]
@DirectXMan12 clear to approve, tyvm!

File: hack/check-everything.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ set -e
 hack_dir=$(dirname ${BASH_SOURCE})
 source ${hack_dir}/common.sh
 
-k8s_version=1.14.1
+k8s_version=1.16.4
```
Comment (AmitKumarDas) [inline hack/check-everything.sh]: Can we mention this version upgrade in the PR message itself. Is CRD with version v1 supported from 1.16.4?

File: pkg/envtest/crd.go [hunk 2]
```diff
@@ -109,27 +111,57 @@ func defaultCRDOptions(o *CRDInstallOptions) {
...
-func WaitForCRDs(config *rest.Config, crds []*apiextensionsv1beta1.CustomResourceDefinition, options CRDInstallOptions) error {
+func WaitForCRDs(config *rest.Config, crds []runtime.Object, options CRDInstallOptions) error {
 	// Add each CRD to a map of GroupVersion to Resource
 	waitingFor := map[schema.GroupVersion]*sets.String{}
-	for _, crd := range crds {
+	for _, crd := range runtimeListToUnstructured(crds) {
 		gvs := []schema.GroupVersion{}
-		if crd.Spec.Version != "" {
-			gvs = append(gvs, schema.GroupVersion{Group: crd.Spec.Group, Version: crd.Spec.Version})
+		crdGroup, _, err := unstructured.NestedString(crd.Object, "spec", "group")
```
Comment (AmitKumarDas) [inline pkg/envtest/crd.go]: Will it be good to check the found option & return err for both spec.group not found as well err cases?

File: hack/check-everything.sh (hunk 1 above)
Comment (vincepri) [reply hack/check-everything.sh]: Added a message

File: pkg/envtest/crd.go (hunk 2 above)
Comment (vincepri) [reply pkg/envtest/crd.go]: I'm ok either way, although it's a change from the previous behavior

**Comment by vincepri** [comment]
@DirectXMan12 @gerred Any updates on the status of this PR?

**Comment by gerred** [comment]
@vincepri I can't approve as a reviewer. @DirectXMan12 @droot @pwittrock are we OK approving here? I don't mind also becoming a CR approver to help here more. :)

**Comment by DirectXMan12** [comment]
/approve

as per the reviewers' LGTMs

This has the nice effect of getting CRD out of our public API surface.

**Comment by vincepri** [comment]
Thanks @DirectXMan12 !

@gerred need another lgtm if you're around :)

**Comment by DirectXMan12** [comment]
/lgtm
(reapplying as per above)

**Pull request merged**